vms/platformvm/txs/executor/proposal_tx_executor.go

```diff
@@ -131,25 +136,22 @@ func (e *ProposalTxExecutor) AddValidatorTx(tx *txs.AddValidatorTx) error {
 
 	txID := e.Tx.ID()
 
+	// Consume the input UTXOs whether we accept commit or abort blocks
+	avax.Consume(e.OnProposalBlockState, tx.Ins)
+
```

Comment:
this probably blurs a bit the distinction of what state changes go in proposal block state and what goes in options block states.
However it should be marginally more efficient to do this once, in proposal block state, rather than twice in options blocks.